proto/archway/rewards/v1beta1/genesis.proto

```diff
@@ -0,0 +1,37 @@
+syntax = "proto3";
+package archway.rewards.v1beta1;
+
+option go_package = "github.com/archway-network/archway/x/rewards/types";
+
+import "gogoproto/gogo.proto";
+import "archway/rewards/v1beta1/rewards.proto";
+
+// GenesisState defines the initial state of the tracking module.
+message GenesisState {
+  // params defines all the module parameters.
+  Params params = 1 [
+    (gogoproto.nullable) = false
+  ];
+  // contracts_metadata defines a list of all contracts metadata.
+  repeated GenesisContractMetadata contracts_metadata = 2 [
+    (gogoproto.nullable) = false
+  ];
+  // block_rewards defines a list of all block rewards objects.
+  repeated BlockRewards block_rewards = 3 [
+    (gogoproto.nullable) = false
+  ];
+  // tx_rewards defines a list of all tx rewards objects.
+  repeated TxRewards tx_rewards = 4 [
+    (gogoproto.nullable) = false
+  ];
+}
+
+// GenesisContractMetadata is used init ContractMetadata state via module genesis.
+message GenesisContractMetadata {
+  // contract_address defines the contract address.
+  string contract_address = 1;
```

Comment:
IDK if to put contract_address info in the metadata itself, even if it would mean redundancy over metadata's primary key and field, the SDK I think does keep this redundancy which I think it's fine if it simplifies protos.

Reply:
Done